rclcpp/include/rclcpp/publisher.hpp

```diff
@@ -109,21 +100,15 @@ class Publisher : public PublisherBase
     // interprocess publish, resulting in lower publish-to-subscribe latency.
     // It's not possible to do that with an unique_ptr,
     // as do_intra_process_publish takes the ownership of the message.
-    uint64_t message_seq;
     bool inter_process_publish_needed =
       get_subscription_count() > get_intra_process_subscription_count();
```

Comment:
I think we have an old error here, consider:
```cpp
bool inter_process_publish_needed =
      (get_subscription_count() > get_intra_process_subscription_count()) ||
      intra_process_is_enabled_;
```

If not, it may break transient local qos policy.

Reply:
I have send a patch for this https://github.com/ros2/rclcpp/pull/786#issue-298548085